test_analysis.py

```
@@ -0,0 +1,43 @@
+import glob
+import re
+
+lib_dir = "lib"
+test_dir = "test"
+
+# Find all dart files in lib excluding .g.dart, .freezed.dart
+lib_files = []
+for root, _, files in os.walk(lib_dir):
+    for file in files:
+        if file.endswith(".dart") and not file.endswith(".g.dart") and not file.endswith(".freezed.dart"):
+            lib_files.append(os.path.join(root, file))
```

Ruff (0.15.7)

[warning] Use `list.extend` to create a transformed list

(PERF401)